src/rtc_peer.ts

```diff
@@ -382,21 +560,57 @@ export class RTCPeer extends EventEmitter {
                 streams: [stream!],
             });
 
-            if (opts?.codec && trx.setCodecPreferences) {
-                this.logger.logDebug('setting video codec preference', opts.codec);
-                trx.setCodecPreferences([opts.codec]);
+            if (trx.setCodecPreferences) {
+                const vp8Codec = await RTCPeer.getVideoCodec(CodecMimeType.VP8);
+                if (!vp8Codec) {
+                    throw new Error('VP8 codec not found');
+                }
+
+                const codecs = [vp8Codec];
+                const av1Codec = await RTCPeer.getVideoCodec(CodecMimeType.AV1);
+                if (av1Codec) {
+                    codecs.push(av1Codec);
+                }
+
+                if (av1Codec && this.config.enableAV1 && this.codecSupportMap[CodecMimeType.AV1] === CodecSupportLevel.Full) {
+                    this.logger.logDebug('RTCPeer.addTrack: AV1 enabled and full support in call, setting AV1 codec as preferred');
+                    codecs.reverse();
```

Comment:
Order is essential for the other side (pion). First codec in the list will be preferred. Why send both, though? That's because we need these transceivers to also support receiving the other codec.